src/models/data.cpp

```diff
@@ -12,6 +12,79 @@
 
 using namespace Data;
 
+ItemCost::ItemCost(std::size_t size)
+    : m_cost(size)
+{
+}
+
+ItemCost::ItemCost(qint64 value, std::size_t count)
+    : m_cost(value, count)
+{
+}
+
+ItemCost::ItemCost(std::valarray<qint64>&& cost)
+    : m_cost(cost)
+{
+}
+ItemCost::~ItemCost() = default;
+
+void ItemCost::resize(std::size_t newSize, qint64 value)
+{
+    m_cost.resize(newSize, value);
+}
+
+std::size_t ItemCost::size() const
+{
+    return m_cost.size();
+}
+
+ItemCost ItemCost::operator+(const ItemCost& rhs) const
+{
+    return static_cast<std::valarray<qint64>>(m_cost + rhs.m_cost);
```

Comment:
out of interest: why the static cast here and below?

Reply:
because the resulting type is `std::_Expr<std::__detail::_BinClos<std::__minus, std::_ValArray, std::_ValArray, long long int, long long int>, long long int>`

Reply:
so I am casting that to `std::valarray` and then the constructor will turn that into `ItemCost`

Reply:
this should work too, no?

```
return ItemCost(m_cost + rhs.m_cost);
```